src/cephadm/cephadm

```diff
@@ -308,32 +313,65 @@ class SNMPGateway:
         return cls(ctx, fsid, daemon_id,
                    get_parm(ctx.config_json), ctx.image)
 
-    def get_version(self):
-        pass
+    @staticmethod
+    def get_version(ctx: CephadmContext, fsid, daemon_id: str) -> Optional[str]:
+        """Return the version of the notifer from it's http endpoint"""
+        path = os.path.join(ctx.data_dir, fsid, f"snmp-gateway.{daemon_id}", 'unit.meta')
+        try:
+            with open(path, 'r') as env:
+                metadata = json.loads(env.read())
+        except (OSError, json.JSONDecodeError):
+            return None
 
-    def get_container_envs(self):
-        if self.snmp_version == 'V2c':
-            return [
-                f'SNMP_NOTIFIER_COMMUNITY={self.snmp_community}'
-            ]
-        else:
-            # TODO for snmpv3 auth
-            return []
+        ports = metadata.get('ports', [])
+        if not ports:
+            return None
+
+        try:
+            with urlopen(f"http://0.0.0.0:{ports[0]}/") as r:
+                html = r.read().decode('utf-8').split('\n')
```

Comment:
You think we can also get this from the container without having any daemons deployed? does the binary have a --version?

Reply:
That would be clever. I'm not sure I follow though. The current get_version methods of the other classes receive a running container id and exec in to it to run the --version command. I'm trying to follow the existing pattern you've defined.